CONTRIBUTING.md

```diff
@@ -1,41 +1,67 @@
 # Contributing
 
+Thanks for your interest in contributing to temBoard. temBoard is an open
+source project open to contribution from idea to code and more.
 
-[dalibo/temboard]: https://github.com/dalibo/temboard
-
-## Reporting an Issue & Submitting a Patch
 
-We use the [dalibo/temboard] project to track issue and review
-contributions. Fork the main repository and open a PR against
-`master` as usual.
+## Submitting an Issue or a Patch
 
+We use the [dalibo/temboard] GitHub project to track issues and review
+contributions. Fork the main repository and open a pull request against
+`master` branch as usual.
 
-## Development Environment Setup
 
-You can quickly set up a dev env with virtualenv and Docker Compose. Running
-development version of UI and agent requires two shells, one for the UI and one
-for the agent.
+## Cloning the Repository
 
-Get temBoard UI and agent sources:
+Get temBoard UI and agent sources in one single repository:
 
 ```console
 $ git clone https://github.com/dalibo/temboard.git
 $ cd temboard/
 ```
 
-Then, create a virtualenv for Python3.6+ or Python2.7, activate it. Then
-install temBoard and pull docker images:
 
-``` console
-$ pip install -e ui/ -r ui/requirements-dev.txt
-...
-$ docker-compose pull
-```
+## Directories Overview
+
+[dalibo/temboard] git repository contains a few sub-projects. Here is a quick
+overview.
+
+- `docs/` - Global mkdocs documentation sources.
+- `ui/` - Python Tornado project for temBoard UI aka server.
+    - `ui/temboardui/toolkit` - Shared library between agent and UI.
+- `agent/` - Bare Python project for temBoard agent.
+    - `agent/temboardagent/toolkit` - Symlink to toolkit in UI source tree.
+- `perfui/` - Docker & Grafana project to visualize temBoard performances traces.
+- `docker/` - Development and quickstart docker files.
+
+Python package for temBoard UI is `temboardui` and `temboardagent` for temBoard
+agent.
+
+
+## Development Requirements
+
+You need the following software to contribute to temBoard:
 
-Then, bootstrap development with `make devenv`.
+- bash, git, make, psql.
+- Docker Compose.
+- Python 3.6.
+- NodeJS and npm for building some assets.
+
+
+## Setup Development
+
+Running development version of UI and agent requires two terminals, one for the
+UI and one for the agent.
+
+The `develop` make target creates a virtual environment for Python 3.6,
+installs temBoard UI, its dependencies, development tools, starts docker
+services and initializes temBoard database.
 
 ``` console
-$ make devenv
+$ make develop
+make venv-3.6
+make[1] : on entre dans le répertoire « /home/.../src/dalibo/temboard »
+python3.6 -m venv .venv-py3.6/
```

Comment:
This requires python3.6-venv to be installed. This was not the case on my system.
`python3.6 -m virtualenv .venv-py3.6/` seem to work. I may not give the same results though.

Reply:
Je crois que Debian a un comportement particulier, venv n'est **pas** livré avec Python, alors que c'est dans la bibiliothèque standard ( https://docs.python.org/fr/3/library/venv.html ), on a le même problème avec `ensurepip` il me semble. J'ai ajouté `venv` dans les prérequis.